Modules/Exercise03/Tests/ProductServiceTest.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Modules\Exercise03\Tests\Service;
+
+use InvalidArgumentException;
+use Modules\Exercise03\Models\Product;
+use Modules\Exercise03\Repositories\ProductRepository;
+use Modules\Exercise03\Services\ProductService;
+use Tests\TestCase;
+
+class ProductServiceTest extends TestCase
+{
+    protected $productService;
+    protected $productRepository;
+
+    public function setUp(): void
+    {
+        parent::setUp(); // TODO: Change the autogenerated stub
+        $this->productRepository = $this->mock(ProductRepository::class);
+        $this->productService = new ProductService($this->productRepository);
+    }
+
+    public function test_calculate_discount_invalid()
+    {
+        $totalProducts = [
+            Product::CRAVAT_TYPE => -1,
+            Product::WHITE_SHIRT_TYPE => null,
+            Product::OTHER_TYPE => null,
+        ];
+        $this->expectException(InvalidArgumentException::class);
+        $this->productService->calculateDiscount($totalProducts);
+    }
+
+    public function test_calculate_discount_cravat()
+    {
+        $totalProducts = [
+            Product::CRAVAT_TYPE => 1,
+            Product::WHITE_SHIRT_TYPE => 0,
+            Product::OTHER_TYPE => 1,
+        ];
+        $result = $this->productService->calculateDiscount($totalProducts);
+        $this->assertEquals($result, 0);
+    }
+
+    public function test_calculate_discount_white_shirt()
+    {
+        $totalProducts = [
+            Product::CRAVAT_TYPE => 0,
+            Product::WHITE_SHIRT_TYPE => 1,
+            Product::OTHER_TYPE => 1,
+        ];
+        $result = $this->productService->calculateDiscount($totalProducts);
+        $this->assertEquals($result, 0);
```

Comment:
sao chỗ này k gọi constant như này mà lại truyền 0 vào em ơi $this->productService::CRAVAT_WHITE_SHIRT_DISCOUNT

Reply:
Không có constant value = 0 đâu anh